x/storage/keeper/keeper.go

```diff
@@ -2321,3 +2414,166 @@ func (k Keeper) SetTag(ctx sdk.Context, operator sdk.AccAddress, grn types2.GRN,
 
 	return nil
 }
+
+func (k Keeper) UpdateObjectContent(
+	ctx sdk.Context, operator sdk.AccAddress, bucketName, objectName string, payloadSize uint64,
+	opts types.UpdateObjectOptions,
+) error {
+	store := ctx.KVStore(k.storeKey)
+
+	bucketInfo, found := k.GetBucketInfo(ctx, bucketName)
+	if !found {
+		return types.ErrNoSuchBucket
+	}
+	err := bucketInfo.CheckBucketStatus()
+	if err != nil {
+		return err
+	}
+	objectInfo, found := k.GetObjectInfo(ctx, bucketName, objectName)
+	if !found {
+		return types.ErrNoSuchObject
+	}
+	// check object status
+	if objectInfo.ObjectStatus != types.OBJECT_STATUS_SEALED {
+		return types.ErrUpdateObjectNotAllowed.Wrapf("The object is not sealed yet")
+	}
+	if objectInfo.IsUpdating {
+		return types.ErrObjectIsUpdating.Wrapf("The object is already being updated")
+	}
+	if objectInfo.SourceType != opts.SourceType {
```

Comment:
I think for object, we dont have to check the source when updating the contend.
Because for object, it is only possible to update contend on Gnfd side, event it is mirrored to BSC/opBNB

Reply:
removed